statsmodels/tsa/tests/test_tsa_indexes.py

```diff
@@ -710,6 +739,105 @@ def test_prediction_increment_pandas_dates_nanosecond():
     assert_equal(prediction_index.equals(desired_index), True)
 
 
+@pytest.mark.skipif(not has_range_index, reason='No RangeIndex')
+def test_range_index():
+    tsa_model.__warningregistry__ = {}
+
+    endog = pd.Series(np.random.normal(size=5))
+    assert_equal(isinstance(endog.index, pd.RangeIndex), True)
+    # Warning should not be given
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        mod = tsa_model.TimeSeriesModel(endog)
+        assert_equal(len(w), 0)
+
+
+@pytest.mark.skipif(not has_range_index, reason='No RangeIndex')
+def test_prediction_rangeindex():
+    index = supported_increment_indexes[2][0]
+    endog = pd.Series(dta[0], index=index)
+    mod = tsa_model.TimeSeriesModel(endog)
+
+    # Basic prediction: [0, end]
```

Comment:
What makes the three cases (Basic prediction, Negative Index, Forecasting) special?  Does this cover the space of Cases That Should Be Tested?  Basically asking for a comment at the top of this test

Reply:
I think these are the three most common use cases, but are not exhaustive.